### PR TITLE
Introduce a Gradle plugin for Realm

### PR DESCRIPTION
**Pull request by emanuelez**
This can substitute having to define the dependency manually.
In the long run this will help us to split Realm and the annotation processor, saving a few KB from installed Realm-based apps.
Also this will allow byte-code manipulation.

### CODE REVIEW

**Comment by emanuelez** [comment]
@cmelchior @kneth @bmunkholm

**Comment by cmelchior** [comment]
You don't need 4 different versions of the launcher icon and we use a Realm logo as app icon for all our example apps, so I would add that.

But apart from that :+1:

File: gradle-plugin/sample/src/androidTest/java/io/realm/example/sample/ApplicationTest.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package io.realm.example.sample;
```
Comment (kneth) [inline gradle-plugin/sample/src/androidTest/java/io/realm/example/sample/ApplicationTest.java]: Standard header is missing.

File: gradle-plugin/sample/src/main/java/io/realm/example/sample/models/Person.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2014 Realm Inc.
```
Comment (kneth) [inline gradle-plugin/sample/src/main/java/io/realm/example/sample/models/Person.java]: 2014 -> 2015

**Comment by kneth** [comment]
Besides my comments: :+1:

**Comment by emanuelez** [comment]
Ready for the second round @cmelchior @kneth @bmunkholm

File: gradle-plugin/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#Mon May 04 12:41:37 CEST 2015
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.2-all.zip
```
Comment (kneth) [inline gradle-plugin/gradle/wrapper/gradle-wrapper.properties]: Are we going for version 2.4?

Comment (emanuelez) [inline gradle-plugin/gradle/wrapper/gradle-wrapper.properties]: Yes, depending on which PR gets merged first, the other will have to adapt

**Comment by bmunkholm** [comment]
:+1:

**Pull request merged**